nestkernel/node_collection.h

```diff
@@ -721,6 +730,28 @@ inline nc_const_iterator nc_const_iterator::operator+( const size_t n ) const
   return it += n;
 }
 
+
+inline size_t nc_const_iterator::operator-( const nc_const_iterator& other ) const
```

Comment:
Do we need this operator? If not, we can remove it. The implementation by explicit iteration is slow. At least we should avoid code duplication by something like this:
```c++
for ( auto it = nc_const_iterator( min( *this, other ) ) ; it < max( *this, other ) ; ++it )
{
  ++i;
}
```